File: src/dymaptic.Blazor.GIS.API.Core/Objects/AddressCandidate.cs
using dymaptic.Blazor.GIS.API.Core.Components.Geometries;


namespace dymaptic.Blazor.GIS.API.Core.Objects;

public class AddressCandidate
{
    public string? Address { get; set; }
    public Dictionary<string, string> Attributes { get; set; } = new();
    public Extent? Extent { get; set; }
    public Point? Location { get; set; }
    public double? Score { get; set; }
}

File: src/dymaptic.GeoBlazor.Core/Scripts/addressCandidate.ts
import {buildDotNetExtent} from "./extent";
import {buildDotNetPoint} from "./point";
import {hasValue} from './geoBlazorCore';

export function buildDotNetAddressCandidate(addressCandidate: any): any {
    if (!hasValue(addressCandidate)) {
        return null;
    }

    return {
        address: addressCandidate.address,
        attributes: addressCandidate.attributes,
        extent: buildDotNetExtent(addressCandidate.extent),
        location: buildDotNetPoint(addressCandidate.location),
        score: addressCandidate.score
    }
}

export async function buildJsAddressCandidate(dotNetObject: any): Promise<any> {
    let {buildJsAddressCandidateGenerated} = await import('./addressCandidate.gb');
    return await buildJsAddressCandidateGenerated(dotNetObject);
}
